x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/state.ts

```diff
@@ -90,14 +91,12 @@ export class State {
             await desiredState.commitChanges(startingState);
             return { status: 'success' as const, changes: desiredState.changes(startingState) };
           } catch (error) {
-            await desiredState.attemptRollback(startingState, error);
-            return {
-              status: 'failed_with_rollback' as const,
-              error: new StatusError(
-                `Failed to apply changes but successfully rolled back to previous state: ${error.message}`,
-                error.statusCode ?? 500
-              ),
-            };
+            // So here I'll capture the error message and send telemetry and then throw that error
+
+            throw new FailedToChangeStateError(
+              `Failed to change state: ${error.message}. The cluster state may be inconsistent. If you experience issues, please use the resync API to restore a consistent state.`,
```

Comment:
Wondering if a `/debug` endpoint that determines whether a stream state is in sync with the es state would make sense. Could be an option in the UI (something like `Experiencing issues ? Check if the state is in sync), or an option that shows when this error happens to check if a resync is actually necessary, or just a troubleshooting tool for sdh

Reply:
Agree, we spoke about it here https://github.com/elastic/streams-program/discussions/139. 
@flash1293 Should we translate that discussion to an issue and ask Luca to give a priority? 
In the discussion we pushed it past M2 to M3 but that's passed, and M4 might already be full of robustness topics so this might now become M5?

Reply:
Before we work further around this failure mode, let's wait for it to become a bigger problem. If it is an issue, we can handle it with debug endpoints and similar, but I still hope we can mitigate it well enough so we don't need this.

For a user it's definitely strange this can even be a thing...